ambry-router/src/test/java/com.github.ambry.router/GetBlobInfoOperationTest.java

```diff
@@ -414,6 +390,26 @@ public void testBlobAuthorizationFailureOverrideAll() throws Exception {
     testErrorPrecedence(serverErrorCodes, RouterErrorCode.BlobAuthorizationFailure);
   }
 
+  /**
+   * Tests the case where all servers return the same server level error code
+   * @throws Exception
+   */
+  @Test
+  public void testFailureOnServerErrors() throws Exception {
+    List<ServerErrorCode> serverErrors = new ArrayList<>(Arrays.asList(ServerErrorCode.values()));
+    // set the status to various server level level errors (remove all partition level errors or non errors)
+    serverErrors.remove(ServerErrorCode.Blob_Deleted);
+    serverErrors.remove(ServerErrorCode.Blob_Expired);
+    serverErrors.remove(ServerErrorCode.No_Error);
+    serverErrors.remove(ServerErrorCode.Blob_Authorization_Failure);
+    serverErrors.remove(ServerErrorCode.Blob_Not_Found);
+    for (ServerErrorCode serverErrorCode : serverErrors) {
+      mockServerLayout.getMockServers().forEach(server -> server.setServerErrorForAllRequests(serverErrorCode));
+      assertOperationFailure(serverErrorCode == ServerErrorCode.Disk_Unavailable ? RouterErrorCode.AmbryUnavailable
```

Comment:
question: if `Replica_Unavailable` is introduced, this type of server error should be mapped to `RouterErrorCode.AmbryUnavailable` as well?

Reply:
Yeah - I think both should map to `AmbryUnavailable`